packages/synchronizer/src/__tests__/fetcher.spec.ts

```diff
@@ -117,4 +119,42 @@ describe('Fetcher Tests', () => {
       assert.match(queryResult.error ?? '', /Connection error/);
     });
   });
+
+  describe('getOriginConfig', () => {
+    it('fetches and parses config.js from origin', async () => {
+      return new Promise((res, rej) => {
+        const app = express();
+
+        app.get('/config.js', (_req, res) => {
+          res.send(`
+            globalThis.import_meta_env = {
+              API_ORIGIN: "https://api.monokle.local",
+              CONTENT_ORIGIN: "https://api.monokle.local",
+              COLLAB_STREAM_URL: "wss://api.monokle.local/collab",
+              COLLAB_HTTP_URL: "https://api.monokle.local/collab",
+
+              OIDC_DISCOVERY_URL: "https://id.monokle.local/realms/monokle",
+              CLIENT_ID: "clientId",
+            };
+          `);
+        });
+
+        const server = app.listen(13000, async () => {
```

Comment:
Starting a server is quite heavy for a unit test. If we keep it at a handful of these tests as full integration tests. But beyond that I'd consider mocking the HTTP Client (i.e. `fetch`) with Jest etc.

Reply:
I agree, with more tests like this we either mock it or start single server instance to be reused by all test. At this stage, entire test run takes less than 500ms so I think we are good for now. But something to keep in mind in the future 👍